docker/install-dynamic-plugins.py

```diff
@@ -789,6 +792,87 @@ def wait_for_lock_release(lock_file_path):
      time.sleep(1)
    print("======= Lock released.")
 
+def extract_catalog_index(catalog_index_image: str, catalog_index_mount: str) -> str:
+    """Extract the catalog index OCI image and return the path to dynamic-plugins.default.yaml if found."""
+    if not catalog_index_image:
+        print("======= No CATALOG_INDEX_IMAGE specified, skipping catalog index extraction", flush=True)
+        return None
+    try:
+        print(f"\n======= Extracting catalog index from {catalog_index_image}", flush=True)
+        skopeo_path = shutil.which('skopeo')
+        if skopeo_path is None:
+            print("WARNING: skopeo executable not found in PATH, skipping catalog index extraction", flush=True)
+            return None
```

Comment:
It should raise an exception in this case rather than a warning message, as this is a fatal error IMO.